src/main/java/org/broadinstitute/consent/http/service/dao/DacServiceDAO.java

```diff
@@ -0,0 +1,68 @@
+package org.broadinstitute.consent.http.service.dao;
+
+import com.google.inject.Inject;
+import java.sql.SQLException;
+import java.util.List;
+import java.util.Objects;
+import org.broadinstitute.consent.http.models.Dac;
+import org.broadinstitute.consent.http.models.DataAccessAgreement;
+import org.broadinstitute.consent.http.util.ConsentLogger;
+import org.jdbi.v3.core.Jdbi;
+import org.jdbi.v3.core.statement.Update;
+
+public class DacServiceDAO implements ConsentLogger {
+
+  private final Jdbi jdbi;
+
+  @Inject
+  public DacServiceDAO(Jdbi jdbi) {
+    this.jdbi = jdbi;
+  }
+
+  public void deleteDacAndDaas(Dac dac, List<DataAccessAgreement> daas)
+      throws IllegalArgumentException, SQLException {
+    // fail fast
+    if (Objects.isNull(dac)) {
+      throw new IllegalArgumentException("Invalid DAC");
+    }
+    jdbi.useHandle(handle -> {
+      handle.getConnection().setAutoCommit(false);
+
+      jdbi.useTransaction(handler -> {
+        final String deleteFromLcDaa = "DELETE FROM lc_daa WHERE daa_id in (SELECT daa_id FROM data_access_agreement WHERE initial_dac_id = :dacId)";
+        final String deleteFromDacDaa = "DELETE FROM dac_daa WHERE dac_id = :dacId";
+        final String deleteFromDaa = "DELETE FROM data_access_agreement WHERE initial_dac_id = :dacId";
+        final String deleteMembers = "DELETE FROM user_role WHERE dac_id = :dacId";
+        final String updateDatasets = "UPDATE dataset SET dac_id = null, needs_approval = true WHERE dac_id = :dacId";
+        final String deleteDac = "DELETE FROM dac where dac_id = :dacId";
+
+        Update lcDaaDeletion = handler.createUpdate(deleteFromLcDaa);
+        lcDaaDeletion.bind("dacId", dac.getDacId());
+        lcDaaDeletion.execute();
+
+        Update dacDaaDeletion = handler.createUpdate(deleteFromDacDaa);
+        dacDaaDeletion.bind("dacId", dac.getDacId());
+        dacDaaDeletion.execute();
+
+        daas.forEach(id -> {
+          Update daaDeletion = handler.createUpdate(deleteFromDaa);
+          daaDeletion.bind("dacId", dac.getDacId());
+          daaDeletion.execute();
+        });
```

Comment:
I think this is incorrect. Imagine if DAAs were passed in with different DAC ids. We should not be deleting DAAs created by other DACs, we should only be deleting the DAAs created by the DAC we're focused on.

Reply:
That's why we are passing in the `filteredDaas` to this method, only the DAAs that were created by this DAC. I was doing the filtering in the service method so that only the deletions would have to happen in the ServiceDAO method.

Reply:
That's why I think this is incorrect. This method has to stand on it's own and not be dependent on some external process to do pre-work before getting here.